src/supertux/menu/profile_menu.cpp

```diff
@@ -91,50 +85,28 @@ ProfileMenu::menu_action(MenuItem& item)
   if (id == 1)
   {
     std::string profile_name = item.get_text();
-    if (profile_name[0] == '[' && profile_name[profile_name.size() - 1] == ']')
-    {
-      profile_name.erase(0, 1);
-      profile_name.erase(profile_name.size() - 1);
-    }
-    g_config->profile = profile_name;
+    //If the selected item's name is equal to the one of the current profile with '[' and ']' from both sides,
+    //the currently selected profile is being selected again, so don't do anything.
+    if ('[' + g_config->profile + ']' != profile_name)
+      g_config->profile = profile_name;
   }
   else if (id == 2 || id == 4)
   {
-    std::string conf_msg;
     const bool is_reset = id == 2;
-    if (is_reset)
-    {
-      conf_msg = "Resetting your profile will delete your game progress on it.\nAre you sure?";
-    }
-    else
-    {
-      conf_msg = "This will delete your profile, wiping all game progress on it and removing it from the list.\nAre you sure?";
-    }
+    std::string conf_msg = is_reset ? "Resetting your profile will delete all of your game progress on it.\nAre you sure?" :
+      "This will delete your profile, wiping all game progress on it and removing it from the list.\nAre you sure?";
     Dialog::show_confirmation(_(conf_msg), [this, is_reset]() {
       delete_savegames(g_config->profile, is_reset);
       if (!is_reset) g_config->profile = "default";
     });
```

Comment:
I will try to review to not overwork Semphris. Problem with this code is that string extraction script for po files will not catch this. You can try this instead:
```
std::string conf_msg = is_reset?_("Resetting your profile will delete all of your game progress on it.\nAre you sure?") :
      _("This will delete your profile, wiping all game progress on it and removing it from the list.\nAre you sure?");
Dialog::show_confirmation(conf_msg, [this, is_reset]() {
...
```